tests/partitions_test.py

```diff
@@ -103,6 +106,33 @@ def test_isolate_primary(network, args):
     return network
 
 
+@reqs.description("Isolate and reconnect primary")
+def test_isolate_and_reconnect_primary(network, args):
+    primary, backups = network.find_nodes()
+    rules = network.partitioner.isolate_node(primary)
+
+    new_primary, new_view = network.wait_for_new_primary(
+        primary, nodes=backups, timeout_multiplier=6
+    )
+    new_tx = reconfiguration.check_can_progress(new_primary)
+
+    # Drop rules to allow primary to reconnect
```

Comment:
It might be better to use the context manager for managing rules. The explicit `drop()` in the previous test was more of an example for the partioning API.